Source/JavaScriptCore/runtime/Options.cpp

```diff
@@ -635,8 +659,8 @@ static inline void disableAllJITOptions()
     Options::useJITCage() = false;
     Options::useConcurrentJIT() = false;
 
-    if (!Options::useInterpretedJSEntryWrappers())
-        Options::useWasm() = false;
+    if (!Options::useInterpretedJSEntryWrappers() && Options::useWasm())
+        disableAllWasmOptions();
```

Comment:
I don't understand either the old check or your version of the check... Why would we want to disable wasm completely without `Options::useInterpretedJSEntryWrappers()`?